src/main/kotlin/com/openlattice/organizations/DatasetController.kt

```diff
@@ -75,16 +70,20 @@ class DatasetController : DatasetApi, AuthorizingComponent {
     @GetMapping(path = [ID_PATH + EXTERNAL_DATABASE_TABLE])
     override fun getExternalDatabaseTables(
             @PathVariable(ID) organizationId: UUID): Set<OrganizationExternalDatabaseTable> {
-        ensureOwnerAccess(AclKey(organizationId))
-        return edms.getExternalDatabaseTables(organizationId)
+        return edms.getExternalDatabaseTables(organizationId).filter { isAuthorized(Permission.READ).test(AclKey(it.id)) }.toSet()
```

Comment:
If there could be many of these tables, I would rather do a bulk authorization check with `AuthorizationManager.accessChecksForPrincipals`.
For referenceyou can take a look at: `EntitySetsController.checkPermissionsForCreate` or `EntitySetsController.getPropertyMetadataForEntitySets`

Reply:
So it's not an all or nothing thing. We only want to return the tables that they do have READ on, not fail the call if they don't have READ permissions on all of them.

Reply:
Oh nvm I see what you mean

Reply:
If you look at the return value/implementation of `accessChecksForPrincipals`, it won't throw an error, but return an `Authorization` object for each requested `AccessCheck`, which has a map `Map<Permission, Boolean> permissions` where if the value is true, the calling user had that Permission.
If you take a look at `EntitySetsController.checkPermissionsForCreate` that is doing the same thing (but for organizations) you need here (with the difference, that that function also throws an error at the end)

```
        // check read on organization
        val organizationIds = entitySets.map { it.organizationId }.toSet()

        val authorizedOrganizationIds = authorizations
                .accessChecksForPrincipals(
                        organizationIds.map { AccessCheck(AclKey(it), READ_PERMISSION) }.toSet(),
                        Principals.getCurrentPrincipals()
                )
                .filter { it.permissions.contains(Permission.READ) }
                .map { it.aclKey[0] }
                .collect(Collectors.toSet())
```
^ this code block does a bulk check for all organization ids and filters the one, where the calling user has read permission

Reply:
> Oh nvm I see what you mean

Saw this too late  :D

Reply:
The in depth explanation is still appreciated lol